memory/src/main/scala/filodb.memory/format/vectors/HistogramVector.scala

```diff
@@ -578,7 +579,10 @@ class SectDeltaHistogramReader(acc2: MemoryReader, histVect: Ptr.U8)
   }
 
   override def apply(index: Int): HistogramWithBuckets = {
-    require(length > 0)
+    if (length <= 0) {
+      throw EmptyHistogramException(s"""length = $length memory=${toHexString(acc, histVect.addr)}""")
```

Comment:
Change this one too. It has to be "requirement failed"

Reply:
Well, why do we need this anyways of we don't log anything, I will say please don't throw new exception, just log and keep the ``require`` as is.  We also need to ensure exceptions are serialized and deserialized over the proto. Check ``ProtoConverters`` for more details.